### PR TITLE
KIALI-728 List Istio Objects on UI

### PR DESCRIPTION
**Pull request by lucasponce**
- Move Istio Mixer -> Istio Config
- Extend Istio Config to support Mixer Rules + Routing objects
- Filtering and ordering by types and names
- Show yaml config on detail view
- Increase test coverage

### CODE REVIEW

**Comment by lucasponce** [comment]
Increased support of Istio objects:
image

Filter by namespace, type or name:
image

Existing details view extended to cover routing objects:
image

image

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  updateNamespaceFilter = () => this.updateFilters(false);
+
+  updateTypeFilter = () => this.updateFilters(true);
+
+  fetchIstioObjectDetails = (props: IstioConfigId) => {
+    API.getIstioConfigDetail(props.namespace, props.objectType, props.object)
+      .then(response => {
+        let istioObjectDetails: IstioConfigDetails = response['data'];
+        console.log('DELETEME istioObjectDetails: ' + JSON.stringify(istioObjectDetails));
```
Comment (jotak) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: DELETE ME :)

File: src/pages/IstioConfigList/IstioConfigListComponent.tsx [hunk 2]
```diff
@@ -6,56 +6,92 @@ import { ActiveFilter, FilterType } from '../../types/NamespaceFilter';
...
-type IstioRuleListComponentState = {
+const istioTypeFilter: FilterType = {
+  id: 'istiotype',
+  title: 'Istio Type',
+  placeholder: 'Filter by Istio Type',
+  filterType: 'select',
+  filterValues: [
+    {
+      id: 'RouteRule',
+      title: 'RouteRule'
```
Comment (jotak) [inline src/pages/IstioConfigList/IstioConfigListComponent.tsx]: I wonder if the labels should be 'RouteRule' or 'Route Rule' (and same for the other types). The latter is more human-friendly.

File: src/pages/IstioConfigList/__tests__/IstioConfigListComponent.test.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      name: 'red4',
+      created_at: 't4',
+      resource_version: 'r4'
+    },
+    {
+      name: 'blue4',
+      created_at: 't4',
+      resource_version: 'r4'
+    }
+  ],
```
Comment (aljesusg) [inline src/pages/IstioConfigList/__tests__/IstioConfigListComponent.test.ts]: This code is so simialr maybe we could have a loop with an increment

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx (hunk 1 above)
Comment (lucasponce) [reply src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: Ops, thanks !

File: src/pages/IstioConfigList/IstioConfigListComponent.tsx (hunk 2 above)
Comment (lucasponce) [reply src/pages/IstioConfigList/IstioConfigListComponent.tsx]: RouteRule or DestinationPolicy, it is how istio uses these names. I'm basically trying to use same terminology to not introduce changes in kiali that may lead some misunderstandings.
(For example, instead to call Mixer Rule, I just called Rule as Istio uses that name).

**Comment by lucasponce** [comment]
@jotak @aljesusg thx for the review.
I think I've addressed latest comments.

File: src/pages/IstioConfigList/__tests__/IstioConfigListComponent.test.ts (hunk 3 above)
Comment (aljesusg) [reply src/pages/IstioConfigList/__tests__/IstioConfigListComponent.test.ts]: Just a comment no block

**Pull request merged**